Spigot-Server-Patches/0437-Dont-recalculate-permissions-on-world-change.patch

```diff
@@ -0,0 +1,62 @@
+From 094c3bc7a4a1d452d20c5c1f260a346a82b8889e Mon Sep 17 00:00:00 2001
+From: cryptite <cryptite@gmail.com>
+Date: Tue, 17 Dec 2019 15:29:44 -0600
+Subject: [PATCH] Dont recalculate permissions on world change
+
+Often, but not 100% of the time, the server will reconstruct and recalculate the Command Map sent to players
+when they change worlds. This is often laggy; worse-so depending on the extent of the tab-completed commands to be sent.
+In our case, the map sent to the player on login is sufficient and is not tossed by the client for world changes.
+
+As a result, we opt not to recalculate this on world change and this prevents a fairly heavy, and consistent main-thread
+lagspike whenever players change worlds, which can be a lot.
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index bce502181..c41927c80 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -653,4 +653,10 @@ public class PaperWorldConfig {
+         disableHopperMoveEvents = getBoolean("hopper.disable-move-event", disableHopperMoveEvents);
+         log("Hopper Move Item Events: " + (disableHopperMoveEvents ? "disabled" : "enabled"));
+     }
++
++    public boolean recalculatePermsOnWorldChange = false;
```

Comment:
I'm iffy on this, this is one of those things which I believe should be opt-in, people use permissions across world, let's not break their setups